### PR TITLE
Allow pytest 3.0 again

### PR DESCRIPTION
**Pull request by blueyed**
Ref: https://github.com/pytest-dev/pytest-cov/issues/257

### CODE REVIEW

File: src/pytest_cov/plugin.py [hunk 1]
```diff
@@ -291,15 +292,26 @@ def pytest_runtest_teardown(self, item):
...
+    else:
+        @compat.hookwrapper
+        def pytest_runtest_call(self, item):
+            if (item.get_marker('no_cover')
+                    or 'no_cover' in getattr(item, 'fixturenames', ())):
+                self.cov_controller.pause()
+                yield
+                self.cov_controller.resume()
+            else:
+                yield
```
Comment (blueyed) [inline src/pytest_cov/plugin.py]: Could also be moved to compat maybe, fix `functools.partial` maybe.

**Comment by ionelmc** [comment]
@blueyed do we really want this? I don't recall anyone really demanding support for older pytest :)

**Comment by blueyed** [comment]
I thought it was easier.
It was via https://github.com/pytest-dev/pytest-cov/issues/257.

**Comment by ionelmc** [comment]
I think we can drop this, there's no demand.

**Pull request closed without merging**